grove/tomography/operator_utils.py

```diff
@@ -0,0 +1,320 @@
+        return self._is_orthonormal
+
+    def all_hermitian(self):
+        """
+        Check if all basis operators are hermitian.
+        """
+        if self._all_hermitian is None:
+            _log.debug("Testing and caching if all basis operator are hermitian")
+            self._all_hermitian = all((is_hermitian(op) for op in self.ops))
+        return self._all_hermitian
+
+    def __iter__(self):
+        """
+        Iterate over tuples of (label, basis_op)
+        """
+        for l, op in zip(self.labels, self.ops):
+            yield l, op
+
+    def product(self, *bases):
+        """
+        Compute the tensor product with another basis.
+
+        :param bases: One or more additional bases to form the product with.
+        :return (OperatorBasis): The tensor product basis as an OperatorBasis object.
+        """
+        if len(bases) > 1:
+            basis_rest = bases[0].product(*bases[1:])
+        else:
+            assert len(bases) == 1
+            basis_rest = bases[0]
+
+        labels_ops = [(b1l + b2l, qt.tensor(b1, b2)) for (b1l, b1), (b2l, b2) in
+                      itertools.product(self, basis_rest)]
+
+        return OperatorBasis(labels_ops)
+
+    def __pow__(self, n):
+        """
+        Create the n-fold tensor product basis.
+
+        :param int n: The number of identical tensor factors.
+        :return: The product basis.
+        :rtype: OperatorBasis
+        """
+        if not isinstance(n, int):  # pragma no coverage
+            raise TypeError("Can only accept an integer number of factors")
+        if n < 1:  # pragma no coverage
+            raise ValueError("Need positive number of factors")
+        if n == 1:
+            return self
+        return self.product(*([self] * (n - 1)))
+
+    def super_basis(self):
+        """
+        Generate the superoperator basis in which the Choi matrix can be represented.
+
+        The follows the definition in [Chow]_
+
+        .. [Chow] Chow et al., 2012, https://doi.org/10.1103/PhysRevLett.109.060501
+
+        :return (OperatorBasis): The super basis as an OperatorBasis object.
+        """
+
+        labels_ops = [(bnl + "^T (x) " + bml, qt.sprepost(bm, bn)) for (bnl, bn), (bml, bm) in
+                      itertools.product(self, self)]
+        return OperatorBasis(labels_ops)
+
+    def project_op(self, op):
+        """
+        Project an operator onto the basis.
+
+        :param qutip.Qobj op: The operator to project.
+        :return: The projection coefficients as a numpy array.
+        :rtype: scipy.sparse.csr_matrix
+        """
+        if not self.is_orthonormal():  # pragma no coverage
+            raise ValueError("project_op only implemented for orthonormal operator bases")
+        return self.basis_transform.H * qt.operator_to_vector(op).data
+
+    def transfer_matrix(self, sop):
+        """
+        Compute the transfer matrix :math:`R_{jk} = \tr[P_j sop(P_k)]`.
```

Comment:
is `sop` a special command or is that a standard term?

Reply:
@jotterbach Not sure if it's standard - it's for superoperator. I'll expand it.